From e229118562b0cdced7de792ad30a98783c564006 Mon Sep 17 00:00:00 2001
From: MengLi <meng.li@windriver.com>
Date: Thu, 19 Jul 2018 17:30:43 +0800
Subject: [PATCH 1339/1345] driver: net: phy: fix the condition for fiber mode

There is wrong condition for fiber mode in marvell_suspend/resume
function because of the not operator mark ('!').
Therefore, remove the '!' to implement correct processing logic.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/phy/marvell.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/phy/marvell.c b/drivers/net/phy/marvell.c
index b4ca2e0..55b8ca9 100644
--- a/drivers/net/phy/marvell.c
+++ b/drivers/net/phy/marvell.c
@@ -1639,7 +1639,7 @@ static int marvell_suspend(struct phy_device *phydev)
 	int err;
 
 	/* Suspend the fiber mode first */
-	if (!(phydev->supported & SUPPORTED_FIBRE)) {
+	if ((phydev->supported & SUPPORTED_FIBRE)) {
 		err = phy_write(phydev, MII_MARVELL_PHY_PAGE, MII_M1111_FIBER);
 		if (err < 0)
 			goto error;
@@ -1673,7 +1673,7 @@ static int marvell_resume(struct phy_device *phydev)
 	int err;
 
 	/* Resume the fiber mode first */
-	if (!(phydev->supported & SUPPORTED_FIBRE)) {
+	if ((phydev->supported & SUPPORTED_FIBRE)) {
 		err = phy_write(phydev, MII_MARVELL_PHY_PAGE, MII_M1111_FIBER);
 		if (err < 0)
 			goto error;
-- 
1.7.9.5

